In [1]:
import pandas as pd
from scipy import stats
import statsmodels.formula.api as smf
import statsmodels.api as sm

In [17]:
df = pd.read_csv("/content/teachers_rating_data (1).csv")
df.head()

prof  Gender  Age Tenure  Evaluation  Students  Beauty CourseLevel
0  Prof71    Male   52    Yes        3.10        66     3.2       Upper
1  Prof92    Male   35     No        4.55        44     3.8       Lower
2   Prof8  Female   39     No        4.51        96     3.7       Upper
3  Prof38  Female   59    Yes        4.41        33     4.3       Upper
4  Prof57  Female   51    Yes        3.16        92     2.7       Upper

# Q1. Regression with T-test: Using the teachers rating data set, does gender affect teaching evaluation rates?

In [18]:
model = smf.ols('Evaluation ~ Gender', data=df).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:             Evaluation   R-squared:                       0.002
Model:                            OLS   Adj. R-squared:                 -0.007
Method:                 Least Squares   F-statistic:                    0.1991
Date:                Mon, 27 Oct 2025   Prob (F-statistic):              0.656
Time:                        14:36:25   Log-Likelihood:                -103.52
No. Observations:                 120   AIC:                             211.0
Df Residuals:                     118   BIC:                             216.6
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept          4.0040      0.071     56.

Based on the regression results (F(1,118) = 0.20, p = 0.656), there is no statistically significant difference in teaching evaluation ratings between male and female instructors.

Male instructors scored about 0.05 points lower on average than female instructors (β = -0.047, p = 0.656). However, this difference is not statistically significant and could be due to random variation rather than a real gender effect. The model explains only about 0.2% of the variance in evaluation scores (R² = 0.002), indicating that gender has no meaningful impact on teaching evaluations in this dataset.

# Q2. Regression with ANOVA: Using the teachers' rating data set, does beauty score for instructors differ by age?

In [19]:
df['age_group'] = pd.cut(df['Age'], bins=[20, 30, 40, 50, 60, 70],
                         labels=['20s', '30s', '40s', '50s', '60s'])
model = smf.ols('Beauty ~ C(age_group)', data=df).fit()
anova_table = sm.stats.anova_lm(model, typ=2)
print(anova_table)


                 sum_sq     df         F    PR(>F)
C(age_group)   2.120163    4.0  0.963411  0.412581
Residual      63.819795  116.0       NaN       NaN


/usr/local/lib/python3.12/dist-packages/statsmodels/base/model.py:1894: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 4, but rank is 3
  warnings.warn('covariance of constraints does not have full '


An ANOVA test was conducted to determine whether beauty scores differ by age group.
The results were not statistically significant, F(4,116) = 0.96, p = 0.413.
This indicates that instructors’ beauty scores do not significantly differ across age groups, suggesting that age group is not a major factor influencing beauty ratings in this sample.

# Q3. Correlation: Using the teachers' rating dataset, Is teaching evaluation score correlated with beauty score?

In [12]:
import statsmodels.api as sm

X = df['Beauty']
y = df['Evaluation']

X = sm.add_constant(X)

model = sm.OLS(y, X).fit()
predictions = model.predict(X)

print(model.summary())


                            OLS Regression Results                            
Dep. Variable:             Evaluation   R-squared:                       0.009
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     1.107
Date:                Mon, 27 Oct 2025   Prob (F-statistic):              0.295
Time:                        13:44:58   Log-Likelihood:                -97.548
No. Observations:                 120   AIC:                             199.1
Df Residuals:                     118   BIC:                             204.7
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.2624      0.263     16.228      0.0

An OLS regression was conducted to examine whether beauty scores predict teaching evaluations.
The model was not statistically significant, F(1,118) = 1.11, p = 0.295, with an R² = 0.009.
This indicates that beauty scores explain less than 1% of the variance in teaching evaluations. The coefficient for beauty (β = -0.071, p = 0.295) was not significant, suggesting that instructors’ beauty scores do not have a meaningful impact on their evaluation ratings in this dataset.